### PR TITLE
feat(sdk): add new ts examples

### PR DESCRIPTION
**Pull request by notrab**
# Description

Please include a summary of the change and which issue is fixed.
Please also include relevant motivation and context.

# Type of change

- [ ] 💔 Breaking
- [ ] 🚀 Feature
- [ ] 🐛 Fix
- [ ] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: templates/blog/package.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
```
Comment (fbjork) [inline templates/blog/package.json:13]: use the latest version?

Comment (notrab) [reply templates/blog/package.json:13]: This was generated by the CLI.

**Comment by fbjork** [comment]
@notrab should rename `name` to `namespace` in the SDL templates now

**Comment by notrab** [comment]
@fbjork I did already for TS config, there's just conflicts now that need resolving that @pimeys is looking into for the SDL ones that moved folders.

File: templates/chatroom/package.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
+  }
+}
```
Comment (yoav-lavi) [inline templates/chatroom/package.json:15]: Newline

File: cli/crates/backend/src/project.rs [hunk 3]
```diff
@@ -113,19 +113,36 @@ pub async fn init(name: Option<&str>, template: Template<'_>) -> Result<(), Back
...
                 )
                 .await?;
             }
+
+            if std::fs::read_dir(&grafbase_path)
+                .expect("We must have a valid directory in this point.")
+                .any(|dir_entry| {
+                    dir_entry
+                        .ok()
+                        .and_then(|de| de.path().extension().map(|ext| ext == "ts"))
```
Comment (yoav-lavi) [inline cli/crates/backend/src/project.rs:135]: Could we rename these to `dir_entry` and `extension`?

Comment (yoav-lavi) [reply cli/crates/backend/src/project.rs:135]: I'd also add a const for ts

File: templates/blog/package.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  }
+}
```
Comment (yoav-lavi) [inline templates/blog/package.json:15]: Newline

File: templates/commerce/package.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
+  }
+}
```
Comment (yoav-lavi) [inline templates/commerce/package.json:15]: Newline

File: templates/graphql-cloudflare-analytics-sdl/grafbase/.env [hunk 6]
```diff
@@ -0,0 +1 @@
+API_TOKEN=
```
Comment (yoav-lavi) [inline templates/graphql-cloudflare-analytics-sdl/grafbase/.env:1]: Newline, do we not want a default value here? Not sure how it'd behave with this

File: templates/graphql-contentful-sdl/grafbase/.env [hunk 7]
```diff
@@ -0,0 +1,2 @@
+CONTENTFUL_API_URL=
+CONTENTFUL_API_TOKEN=
```
Comment (yoav-lavi) [inline templates/graphql-contentful-sdl/grafbase/.env:2]: Newline, see comments above

File: templates/graphql-contentful/grafbase/.env [hunk 8]
```diff
@@ -1,3 +1,2 @@
-CONTENTFUL_SPACE_ID=
-CONTENTFUL_ENVIRONMENT=
+CONTENTFUL_API_URL=
 CONTENTFUL_API_TOKEN=
```
Comment (yoav-lavi) [inline templates/graphql-contentful/grafbase/.env:2]: Newline, see comments above

File: templates/graphql-contentful/package.json [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
+  }
+}
```
Comment (yoav-lavi) [inline templates/graphql-contentful/package.json:15]: Newline

File: templates/graphql-github-sdl/grafbase/.env [hunk 10]
```diff
@@ -0,0 +1 @@
+GITHUB_ACCESS_TOKEN=
```
Comment (yoav-lavi) [inline templates/graphql-github-sdl/grafbase/.env:1]: Newline, see comments above

File: templates/graphql-github/package.json [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
+  }
+}
```
Comment (yoav-lavi) [inline templates/graphql-github/package.json:15]: Newline

File: templates/graphql-saleor-sdl/grafbase/.env [hunk 12]
```diff
@@ -0,0 +1,2 @@
+ENVIRONMENT_DOMAIN=
+ACCESS_TOKEN=
```
Comment (yoav-lavi) [inline templates/graphql-saleor-sdl/grafbase/.env:2]: Newline, see comments above

File: templates/graphql-saleor/grafbase/.env [hunk 13]
```diff
@@ -1,2 +1,2 @@
-SHOPIFY_STORE_NAME=
-SHOPIFY_STOREFRONT_ACCESS_TOKEN=
\ No newline at end of file
+ENVIRONMENT_DOMAIN=
+ACCESS_TOKEN=
```
Comment (yoav-lavi) [inline templates/graphql-saleor/grafbase/.env:2]: Newline, see comments above

File: templates/graphql-saleor/package.json [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
+  }
+}
```
Comment (yoav-lavi) [inline templates/graphql-saleor/package.json:15]: Newline

File: templates/graphql-shopify-storefront-sdl/grafbase/.env [hunk 15]
```diff
@@ -0,0 +1,2 @@
+SHOPIFY_STORE_NAME=
+SHOPIFY_STOREFRONT_ACCESS_TOKEN=
```
Comment (yoav-lavi) [inline templates/graphql-shopify-storefront-sdl/grafbase/.env:2]: Newline, see comments above

File: templates/graphql-shopify-storefront/package.json [hunk 16]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
+  }
+}
```
Comment (yoav-lavi) [inline templates/graphql-shopify-storefront/package.json:15]: Newline

File: templates/hackernews/package.json [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
+  }
+}
```
Comment (yoav-lavi) [inline templates/hackernews/package.json:15]: Newline

File: templates/openapi-github/package.json [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
+  }
+}
```
Comment (yoav-lavi) [inline templates/openapi-github/package.json:15]: Newline

File: templates/openapi-mux/package.json [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
+  }
+}
```
Comment (yoav-lavi) [inline templates/openapi-mux/package.json:15]: Newline

File: templates/openapi-neon-sdl/grafbase/.env [hunk 20]
```diff
@@ -0,0 +1 @@
+NEON_API_KEY=
```
Comment (yoav-lavi) [inline templates/openapi-neon-sdl/grafbase/.env:1]: Newline, see comments above

File: templates/openapi-neon/package.json [hunk 21]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
+  }
+}
```
Comment (yoav-lavi) [inline templates/openapi-neon/package.json:15]: Newline

File: templates/openapi-openai/package.json [hunk 22]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
+  }
+}
```
Comment (yoav-lavi) [inline templates/openapi-openai/package.json:15]: Newline

File: templates/openapi-stripe/package.json [hunk 23]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
+  }
+}
```
Comment (yoav-lavi) [inline templates/openapi-stripe/package.json:15]: Newline

File: templates/openapi-tinybird-sdl/grafbase/.env [hunk 24]
```diff
@@ -0,0 +1,2 @@
+TINYBIRD_API_SCHEMA=
+TINYBIRD_API_TOKEN=
```
Comment (yoav-lavi) [inline templates/openapi-tinybird-sdl/grafbase/.env:2]: Newline, see comments above

File: templates/openapi-tinybird/package.json [hunk 25]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
+  }
+}
```
Comment (yoav-lavi) [inline templates/openapi-tinybird/package.json:15]: Newline

File: templates/todo/package.json [hunk 26]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
+  }
+}
```
Comment (yoav-lavi) [inline templates/todo/package.json:15]: Newline

File: templates/twitter/package.json [hunk 27]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "devDependencies": {
+    "@grafbase/sdk": "~0.0.20"
+  }
+}
```
Comment (yoav-lavi) [inline templates/twitter/package.json:15]: Newline

File: templates/blog/package.json (hunk 1 above)
Comment (yoav-lavi) [reply templates/blog/package.json:13]: @pimeys do we want to set a specific version and updated the templates on each release? Sounds like a lot

Comment (pimeys) [reply templates/blog/package.json:13]: no we do `~0.1.0` now which matches every version until 1.0

**Review by fbjork — APPROVED** [review]

**Pull request merged**